bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java

```diff
@@ -160,7 +235,33 @@ public void registerParticipation(String gTransactionId, String transactionBlock
      */
     //TODO:Comment for now, might need it for distributed transactions.
     public boolean prepare(String transactionId, String transactionBlockId) {
-          return true;
+        if (transactionManagerEnabled) {
+            return true;
+        }
+        String combinedId = generateCombinedTransactionId(transactionId, transactionBlockId);
+        List<BallerinaTransactionContext> txContextList = resourceRegistry.get(combinedId);
+        if (txContextList != null) {
+            for (BallerinaTransactionContext ctx : txContextList) {
+                try {
+                    XAResource xaResource = ctx.getXAResource();
+                    if (xaResource != null) {
+                        Xid xid = xidRegistry.get(combinedId);
+                        xaResource.prepare(xid);
+                    }
+                } catch (Throwable e) {
+                    log.error("error in prepare the transaction, " + combinedId + ":" + e.getMessage(), e);
+                    return false;
```

Comment:
Why don't we throw this exception? Any reason to handle it and return `falase`?

Reply:
Again, error message does not looks good. Shall we change it as "Error at transaction prepare phase" ?

Reply:
Also, whether printing combinedid makes sense? Don't we need to print transaction id and transaction block id separately?

Reply:
@lasinicl Need this change in other places as well.

Reply:
Fixed

Reply:
TransactionBlockId is something internal right? it is ok to print it in error log? May be ok as a debug log.

Reply:
Fixed